### PR TITLE
Remove accounts hack and correctly restore accounts store counts

### PR DESCRIPTION
**Pull request by sakridge**
#### Problem

Hack introduced to fix accounts code from cleaning up needed stores after they are recovered from a snapshot.

#### Summary of Changes

Restore all accounts versions in the append-vecs to prevent shielded accounts from being cleaned up.

Fixes #

### CODE REVIEW

**Comment by mvines** [comment]
I still get a bank hash verification error when I:
1. Boot a v1.0 node from an existing TdS snapshot with this PR
2. Let the v1.0 node create a new snapshot 
3. Reboot the v1.0 node with the new snapshot

File: runtime/src/accounts_db.rs [hunk 1]
```diff
@@ -1399,34 +1392,51 @@ impl AccountsDB {
...
+                        accounts_index.insert(
+                            *slot_id,
+                            pubkey,
+                            account_info.clone(),
+                            &mut _reclaims,
+                        );
+                    }
                 }
+                // This is important because older write versions otherwise will be lost.
+                accounts_index.roots.insert(*slot_id);
```
Comment (ryoqun) [inline runtime/src/accounts_db.rs]: @sakridge I think you need to add all slots into the root at last outer this loop like this: https://github.com/solana-labs/solana/pull/8337/commits/4c3f293ed9ff2d544f6dac7dabdb86fc6497a079

File: runtime/src/accounts_db.rs [hunk 2]
```diff
@@ -2339,9 +2351,15 @@ pub mod tests {
...
+        accounts.update_accounts_hash(4, &HashMap::default());
+
         let (accounts, skip_account_assertion) = f(accounts, current_slot);
 
-        assert_eq!(4, accounts.accounts_index.read().unwrap().roots.len());
+        print_accounts("post_f", &accounts);
+
+        accounts.verify_bank_hash(4, &HashMap::default()).unwrap();
+
         if skip_account_assertion {
```
Comment (ryoqun) [inline runtime/src/accounts_db.rs]: nits: I think this guard can be removed as well to enable following assertions for https://github.com/solana-labs/solana/pull/8569/files#diff-2099c5256db4eb5975c8834af38f6456R2391

Comment (sakridge) [reply runtime/src/accounts_db.rs]: yea. I'll remove.

**Comment by ryoqun** [comment]
@sakridge For #8436,  I've flushed all previous AccountsDB-related patched to both `v1.0` and `v0.23`. So, rebasing and back-porting this PR should be tedious and straightforward task.

However, considering #8593, I'm open to take this over for you to land far and wide. :)

**Comment by sakridge** [comment]
> 
> 
> @sakridge For #8436, I've flushed all previous AccountsDB-related patches to both `v1.0` and `v0.23`. So, rebasing and back-porting this PR should be tedious and straightforward task.
> 
> However, considering #8593, I'm open to take this over for you to land far and wide. :)

Yea, if you can take this over, that would be great.

**Pull request merged**